Assets/Scripts/Utility/TextProvider.cs

```diff
@@ -267,6 +267,9 @@ public virtual TextFile.Token[] GetRandomTokens(int id)
             // Select a random token stream
             int index = UnityEngine.Random.Range(0, tokenStreams.Count);
 
+            // Select the next to last item from the array if the length of the last one is zero
+            index = (tokenStreams[index].Length == 0 ? index - 1 : index);
```

Comment:
I'm not overly familiar with faction dialogue. Could you give me an example of the text IDs where this is an issue that needed a fix?

As this change would be universal to all random tokens, are there any cases elsewhere where it's possible for this to end up as -1 (e.g. index 0 also has a length of 0).

Reply:
The error would occur for record ID 871 (860 + 11 (B. of Mara info)). The last tokenStreams array item would contain a blank array. Therefore, expandRandomTextRecord errors out because it tries to return tokens[0].text which doesn't exist.

It's possible in theory that a text record could have a blank first token but that would cause an out-of-range exception even without my fix. So far I've only seen a blank last item in this one case with faction dialogue. As long as there's not some strange malformed set of tokens buried deep in TEXT.RSC that has a blank first item then we should be ok.